From: Tobias Geerinckx-Rice <tobias.geerinckx.rice@gmail.com>
Date: Tue, 2 Jun 2015 13:04:19 +0200
Subject: [PATCH] Run `nix-env` & `nix-shell` with `--show-trace`

Evaluation is broken often enough that this seems wise.
Debugging such errors without a backtrace is most unpleasant.

diff -Naur nix-nox-0.0.1/nox/nixpkgs_repo.py nix-nox-0.0.1b/nox/nixpkgs_repo.py
--- nox/nox/nixpkgs_repo.py	2014-10-26 21:50:33.000000000 +0100
+++ nox/nox/nixpkgs_repo.py	2015-07-03 04:46:08.114975479 +0200
@@ -75,7 +75,7 @@
 
 def packages(path):
     """List all nix packages in the repo, as a set"""
-    output = subprocess.check_output(['nix-env', '-f', path, '-qaP', '--drv-path'],
+    output = subprocess.check_output(['nix-env', '-f', path, '-qaP', '--drv-path', '--show-trace'],
                                      universal_newlines=True)
     return set(output.split('\n'))
 
diff -Naur nix-nox-0.0.1/nox/search.py nix-nox-0.0.1b/nox/search.py
--- nox/nox/search.py	2014-09-20 14:55:33.000000000 +0200
+++ nox/nox/search.py	2015-07-03 04:46:54.264813143 +0200
@@ -10,7 +10,7 @@
 
 def nix_packages_json():
     click.echo('Refreshing cache')
-    output = subprocess.check_output(['nix-env', '-qa', '--json'],
+    output = subprocess.check_output(['nix-env', '-qa', '--json', '--show-trace'],
                                      universal_newlines=True)
     return json.loads(output)
 
@@ -70,7 +70,7 @@
                                         value_proc=parse_input)
         attributes = [p.attribute for p in packages]
         if action == 'install':
-            subprocess.check_call(['nix-env', '-iA'] + attributes)
+            subprocess.check_call(['nix-env', '-iA', '--show-trace'] + attributes)
         elif action == 'shell':
             attributes = [a[len('nixpkgs.'):] for a in attributes]
-            subprocess.check_call(['nix-shell', '-p'] + attributes)
+            subprocess.check_call(['nix-shell', '-p', '--show-trace'] + attributes)
